### PR TITLE
[SYSE-180] Tyk-GW / Add proha to api-test wflow

### PR DESCRIPTION
**Pull request by ermirizio**
Add pro-ha flavour to api-tests matrix

### CODE REVIEW

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit:  
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6006019676)

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -187,8 +187,16 @@ jobs:
       matrix:
         conf: [ "sha256", "murmur64" ]
         db: [ "mongo44", "postgres15" ]
+        flavour: [ "pro", "pro-ha" ]
```
Comment (alephnull) [inline .github/workflows/release.yml:190]: I think we just need pro-ha here. At least for PRs.

**Review by alephnull — CHANGES_REQUESTED** [review]
I think we can use pro-ha.yml exclusively.

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: a6967c786754acecd3b7a9ad1612b7469071a927 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6020269395)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: c6e2bec65b3ff59d91395644a059698ac8aef2af 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6025150465)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: 1f4cb84fb1c22e3ffc5dec752822c1afa757cbfc 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6029420101)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: 551ebc36452c316a9ce715413582c939ca9a18b6 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6040216334)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: f13509f5dbdac0c44d33a2c4982700819529e38e 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6040231861)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: 1fb17f4b95acc42a7737c91e7887ca53f2e80234 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6050263061)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: 0e8a4903242169fe48676fd708197f5bd17f0eea 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6050485877)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: 933e7a4a4ab7504b2af36d73e7bfd697b89b85b9 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6051046300)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: 03d3dea4fafe0e1f2ead8f9b0baf29693d8f5378 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6086269652)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: 95950e588f722689adbc91e9c2f1dff25593aa55 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6150680908)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: d75f29268e5d102b6c81dfb2089d34c4c7034dbe 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6150993597)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/5477/merge
Commit: 1a4599482e6b3ea73ea9c004536dae27af5f6249 
Triggered by: pull_request (@ermirizio)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/6151116676)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -187,6 +187,7 @@ jobs:
       matrix:
         conf: [ "sha256", "murmur64" ]
         db: [ "mongo44", "postgres15" ]
+        flavour: [ "pro-ha" ]
```
Comment (alephnull) [inline .github/workflows/release.yml:190]: I don't think this is required at this point. `release.yml` runs for PRs and I feel it is overkill to run this for more than one flavour.

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -243,21 +249,27 @@ jobs:
           tyk_pump_image=\${registry}/tyk-pump:${pump_image_tag}
           # override default with just built tag
           tyk_image=${tags[0]}
+          tyk_alfa_image=${tags[0]}
+          tyk_beta_image=${tags[0]}
           # base dir for config files
-          confs_dir=./pro
+          confs_dir=./"${{ matrix.flavour }}"
```
Comment (alephnull) [inline .github/workflows/release.yml:255]:
```suggestion
          confs_dir=./pro-ha
```
This env can be used in relevant repos for PRs.

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -243,21 +249,27 @@ jobs:
...
+          docker compose -p auto -f "${{ matrix.flavour }}".yml -f deps_"${{ matrix.flavour }}".yml -f ${{ matrix.db }}.yml --env-file versions.env --profile master-datacenter up --quiet-pull -d
+          ./dash-bootstrap.sh http://localhost:3000
+          docker compose -p auto -f "${{ matrix.flavour }}".yml -f deps_"${{ matrix.flavour }}".yml -f ${{ matrix.db }}.yml --env-file versions.env --profile slave-datacenter up --quiet-pull -d
 
       - name: Run tests
+        env:
+          auto_dir: ${{ steps.env_up.outputs.auto_dir	 }}
         run: |
-          docker run --rm --network auto_default \
+          docker run --rm --network auto_default --env-file ${auto_dir}/pytest.env \
```
Comment (alephnull) [inline .github/workflows/release.yml:270]: I think it is better to add the contents of pytest.env into pro-ha.yml.

**Review by alephnull — CHANGES_REQUESTED** [review]

**Comment by ermirizio** [comment]
Closing this in favour of https://github.com/TykTechnologies/tyk/pull/5553 , all changes have been ported there.

**Pull request closed without merging**